### PR TITLE
Lasers passover objects like projectiles unless the target is clicked on

### LINKED ISSUE #28494: Pulse rifle projectiles collide with crit characters
## Description
The projectiles of the pulse rifle will still collide with the bodies of people who are critical, even when not aiming at them.

**Reproduction**
Step 1. Crit someone
Step 2. Fire past them with a pulse rifle

**Screenshots**

**Additional context**
None

### PR DESCRIPTION
**Pull request by Cojoke-dot**
## About the PR

Makes lasers function like projectiles and pass over objects
## Why / Balance

Create consistency within ranged weapons. Fixes #28494
## Technical details

gaaahhh... adds a check when selecting the result of a raycast when it creates the laser that checks if the object should be hit or not.
## Media

https://github.com/space-wizards/space-station-14/assets/83733158/d4f8b698-6d79-44a8-a35d-d8f44f13ba88

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

Uhh... I sure hope none
**Changelog**

:cl:
- tweak: Lasers now pass over things unless clicked like projectiles

### CODE REVIEW

**Comment by TraumaStationOfficial** [comment]
Why does the Crate Reflect?

**Comment by FungiFellow** [comment]
Why does the Crate Reflect?

**Comment by Cojoke-dot** [comment]
That is just something that crates do, they have done that forever. No clue why they are like that but this pr does not add or touch crates.

**Comment by Cojoke-dot** [comment]
> Why does the Crate Reflect? Also thank you for killing crate shields

That is just something that crates do, they have done that forever. No clue why they are like that but this pr does not add or touch crates.

**Comment by Dutch-VanDerLinde** [comment]
fixes https://github.com/space-wizards/space-station-14/issues/27988

File: Content.Server/Weapons/Ranged/Systems/GunSystem.cs [hunk 1]
```diff
@@ -202,6 +203,18 @@ private void OnBallisticPrice(EntityUid uid, BallisticAmmoProviderComponent comp
...
+                            foreach (var colide in rayCastResults)
+                            {
+                                if (colide.HitEntity != gun.Target &&
+                                    CompOrNull<RequireProjectileTargetComponent>(colide.HitEntity)?.Active == true)
+                                    continue;
+
+                                result = colide;
+                                break;
+                            }
+
```
Comment (metalgearsloth) [inline Content.Server/Weapons/Ranged/Systems/GunSystem.cs:217]:
```suggestion

                            // Checks if the laser should pass over unless targeted by its user
                            foreach (var collide in rayCastResults)
                            {
                                if (collide.HitEntity != gun.Target &&
                                    CompOrNull<RequireProjectileTargetComponent>(colide.HitEntity)?.Active == true)
                                {
                                    continue;
                                }

                                result = collide;
                                break;
                            }

```

**Pull request merged**